setup/www/resources/nodejs.org

```diff
@@ -0,0 +1,122 @@
+}
+
+server {
+    listen 80;
+    listen [::]:80;
+    server_name doc.nodejs.org docs.nodejs.org;
+    rewrite ^/(v[0-9]+\.[0-9]+\.[0-9]+)(/?.*)$ https://nodejs.org/docs/$1$2 permanent;
+    rewrite /(.*)$ https://nodejs.org/docs/latest/$1 permanent;
+}
+
+server {
+    listen 80;
+    listen [::]:80;
+    server_name api.nodejs.org;
+    rewrite ^/(v[0-9]+\.[0-9]+\.[0-9]+)(/?.*)$ https://nodejs.org/docs/$1/api$2 permanent;
+    rewrite /(.*)$ https://nodejs.org/docs/latest/api/$1 permanent;
+}
+
+server {
+    listen 80;
+    listen [::]:80;
+    server_name dist.nodejs.org;
+    rewrite /(.*)$ http://nodejs.org/dist/$1 permanent;
+}
+
+server {
+    listen 443 default_server ssl spdy;
+    listen [::]:443 default_server ipv6only=on ssl spdy;
+
+    server_name nodejs.org;
+
+    ssl_certificate ssl/nodejs_chained.crt;
+    ssl_certificate_key ssl/nodejs.key;
+    ssl_trusted_certificate ssl/nodejs_chained.crt;
+    ssl_dhparam ssl/dhparam.pem;
+
+    ssl_ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS;
+    ssl_prefer_server_ciphers on;
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_timeout 24h;
+
+    ssl_stapling on;
+    ssl_stapling_verify on;
+
+    spdy_keepalive_timeout 300;
+    spdy_headers_comp 9;
+
+    keepalive_timeout 60;
+    server_tokens off;
+
+    resolver 8.8.4.4 8.8.8.8 valid=300s;
+    resolver_timeout 10s;
+
+    add_header Strict-Transport-Security max-age=63072000;
+    add_header X-Frame-Options DENY;
+    add_header X-Content-Type-Options nosniff;
+
+    access_log /var/log/nginx/nodejs.org-access.log;
+    error_log /var/log/nginx/nodejs.org-error.log;
+
+    gzip on;
+    gzip_static on;
+    gzip_disable "MSIE [1-6]\.";
+    default_type text/html;
+    gzip_types text/plain text/css application/x-javascript text/xml application/xml application/xml+rss text/javascript;
+
+    if ($host ~* ^www\.){
+        rewrite ^(.*)$ https://nodejs.org$1;
+    }
+
+    location ~ \.json$ {
+      add_header access-control-allow-origin *;
+    }
+
+    location ~ /(dist|docs)/v[0-9]+\.[0-9]+\.[0-9]+/.*\.(tar\.gz|msi|exe|pdb|lib|html|json) {
+      expires max;
+    }
+
+    location /documentation/ {
+      rewrite ^/documentation/api(.*)$ /api$1 permanent;
+    }
+
+    location / {
+        root /home/nodejs/www;
+        index index.html;
+        default_type text/plain;
+        rewrite ^/$ /en/ redirect;
+    }
+
+    location /download {
+        alias /home/dist/public;
+        autoindex on;
+        default_type text/plain;
+    }
+
+    location /dist {
+      alias /home/dist/public/release/;
+      autoindex on;
+      default_type text/plain;
+    }
+
+    location /api {
+      alias /home/dist/public/release/latest/doc/api;
+      autoindex on;
+      default_type text/plain;
+    }
+
+    location /download-stats.json {
+      alias /home/iojs/download-stats.json;
```

Comment:
ahh, yes, actually I need to remove that because it's not workable with CloudFlare anyway